### ISSUE TITLE
Deleting the selected range may produce am invalid file

### ISSUE THREAD

**Issue by Pankraty** [ClosedXML/ClosedXML#1066]
## Read and complete the full issue template

**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature

**Version of ClosedXML**

e.g. 0.94-rc1

**What is the current behavior?**

If the worksheet contains a few selected ranges and one of them becomes invalid (e.g. containing row was deleted) Excel complains that file contains invalid contents. Such file cannot be opened with ClosedXML too.

**What is the expected behavior or new feature?**

1. Invalid ranges should be excluded from the `SelectedRanges` collection
2. If the collection of SelectedRanges in the file contains invalid ranges they should be omitted.

**Did this work in previous versions of our tool?  Which versions?**

No

## Reproducibility

**Code to reproduce problem:**
```c#
        [Test]
        public void InvalidSelectedRangeExcluded()
        {
            using (var wb = new XLWorkbook())
            {
                var ws = wb.AddWorksheet("Sheet1");
                var range1 = ws.Range("B2:C2");
                var range2 = ws.Range("B4:C4");
                ws.SelectedRanges.Clear();

                ws.SelectedRanges.Add(range1);
                ws.SelectedRanges.Add(range2);

                ws.Row(4).Delete();

                Assert.IsFalse(range2.RangeAddress.IsValid);
                Assert.AreEqual(range1, ws.SelectedRanges.Single());
            }
        }

        [Test]
        public void CanLoadFileWithInvalidSelectedRanges()
        {
            using (var stream = TestHelper.GetStreamFromResource(TestHelper.GetResourcePath(@"Misc\InvalidSelectedRange.xlsx")))
            using (var wb = new XLWorkbook(stream))
            {
                var ws = wb.Worksheet(1);

                Assert.AreEqual(1, ws.SelectedRanges.Count);
                Assert.AreEqual("B2:C2", ws.SelectedRanges.Single().RangeAddress.ToString());
            }
        }

```
- [x] I attached a sample spreadsheet.  (You can drag files on to this issue)
[InvalidSelectedRange.xlsx](https://github.com/ClosedXML/ClosedXML/files/2569402/InvalidSelectedRange.xlsx)

**Comment by Pankraty** [comment]
This causes one test to fail in ClosedXML.Report. I am working on it.

**Comment by igitur** [comment]
This is fixed in 0.94, right? I'll reopen if not..
